src/background.js

```diff
@@ -28,6 +33,32 @@ browser.browserAction.onClicked.addListener(() => {
     openOverview()
 })
 
+
+browser.bookmarks.onCreated.addListener((id, bookmarkInfo) => {
+    //TO-DO Add method to handle indexing of bookmark
+    //bookmarkInfo.url gives the URL of the bookmark
+    //id is the unique id of the bookmark
+    //https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/bookmarks/onCreated
+    //Check if there already exists a page with this URL.
+    const samePageCandidates = (await findPagesByUrl({url})).rows.map(row => row.doc);
+    const reusablePage = await tryReidentifyPage({null, bookmarkInfo.url, samePageCandidates})
+    const pageData = fetchPageData(bookmarkInfo.url)
```

Comment:
Yeah, still make the page doc but only with `url`, `title`, and `timestamp` + `id` data from the `BookmarkTreeNode` as the doc `_id`'s `timestamp` and `nonce` values, respectively. The `generatePageDocId` from `src/page-storage` can help you create the `_id`.